containers/src/Data/Sequence/Internal.hs

```diff
@@ -2461,6 +2465,26 @@ update i x (Seq xs)
   | fromIntegral i < (fromIntegral (size xs) :: Word) = Seq (updateTree (Elem x) i xs)
   | otherwise   = Seq xs
 
+-- | \( O(\log(\min(i,n-i))) \). Lookup and delete.
+-- If the index is found, the value corresponding to it is returned and removed
+-- from the sequence. If the index is not found, 'Nothing' is returned together
+-- with the original sequence.
+--
+-- @since 0.6.5
+pop :: Int -> Seq a -> (Maybe a, Seq a)
```

Comment:
I don't like this at all, tradition be damned. Put the `Maybe` on the outside.

Reply:
Likewise for the `IntMap` / `Map` versions?